src/components/Admin/Authentication/SignUp.tsx

```diff
@@ -9,25 +15,133 @@ import {UnauthenticatedPageProps} from '../UnauthenticatedAdmin';
 const SignUp: ({setIsNewUser}: UnauthenticatedPageProps) => JSX.Element = ({
   setIsNewUser,
 }: UnauthenticatedPageProps) => {
-  const [errorGoogle, setErrorGoogle] = useState('');
+  const [googleError, setGoogleError] = useState('');
   const [isLoadingGoogle, setIsLoadingGoogle] = useState(false);
 
+  const [email, setEmail] = useState('');
+  const [emailError, setEmailError] = useState('');
+
+  const [password, setPassword] = useState('');
+  const [showPassword, setShowPassword] = useState(false);
+  const [passwordError, setPasswordError] = useState('');
+
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [showConfirmPassword, setShowConfirmPassword] = useState(false);
+  const [confirmPasswordError, setConfirmPasswordError] = useState('');
+
+  const [generalEmailError, setGeneralEmailError] = useState('');
+  const [isLoadingEmail, setIsLoadingEmail] = useState(false);
+
+  /**
+   * Signs up a user with email in Firebase and handles any errors
+   * @param event - submit form event
+   */
+  async function handleSignUpWithEmail(
+    event: React.FormEvent<HTMLFormElement>
+  ) {
+    // Prevents submission before call to Firebase is complete
+    event.preventDefault();
+
+    setIsLoadingEmail(true);
+    if (password !== confirmPassword) {
+      setConfirmPasswordError('Passwords do not match');
+      setPassword('');
+      setConfirmPassword('');
+      setIsLoadingEmail(false);
+    } else {
+      try {
+        await firebaseAuth.createUserWithEmailAndPassword(email, password);
+      } catch (error) {
+        // Error codes from Firebase documentation
+        switch (error.code) {
+          case 'auth/email-already-in-use': {
+            setEmailError(
+              'Email already has an account. Please sign in or use a different email'
```

Comment:
More than 80 characters (but less than 90 if we want to use a 90 cutoff instead)